Documents DATA_MODEL ui-related methods

mockoocy opened the pull request:
The documentation section, about `QueueEntry.DATA_MODEL` was missing information about `update_dependent_fields` and `ui_schema` methods. Also it gives insight about `ui:schema.col` option that is introduced in https://github.com/mxcube/mxcubeweb/pull/1657

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+It should be used when value of some field stored in the form is
+**dependent** on values of other field (in that case such input would be usually hidden).
+
+The `ui_schema` method returns a python dictionary containing a json string that describes how the form is displayed.
```
```suggestion
The `ui_schema` method returns a Python dictionary containing a JSON string that describes how the form is displayed.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+-   `user_collection_parameters`: Collection parameters relevant to the user complements `collection_parameters`
+-   `legacy_parameters`: Parameters that are still beeing passed arround in the application but not used (for backward compatability)
+
+There are also additional, static, methods, that are used by `mxcube-web` frontend; `update_dependent_fields` and `ui_schema`.
```
```suggestion
There are also additional static methods that are used by MXCuBE-Web frontend: `update_dependent_fields` and `ui_schema`.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+There are also additional, static, methods, that are used by `mxcube-web` frontend; `update_dependent_fields` and `ui_schema`.
+
+The `update_dependent_fields` method is called when form state is updated on the frontend.
+It's goal is to return new values for some form fields.
```
```suggestion
Its goal is to return new values for some form fields.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+**dependent** on values of other field (in that case such input would be usually hidden).
+
+The `ui_schema` method returns a python dictionary containing a json string that describes how the form is displayed.
+It's shape is defined in the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs/version-5.24.10/api-reference/uiSchema) of `rjsf` library that is used to render generic task forms on the `mxcube-web` frontend.
```
```suggestion
Its shape is defined in the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs/version-5.24.10/api-reference/uiSchema) of the `rjsf` library that is used to render generic task forms on the MXCuBE-Web frontend.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+class MyDataModel(BaseModel):
+    @staticmethod
+    def ui_schema():
+        return json.dumps(
+            {
+                "ui:order":[
+                    "field1",
+                    "field2",
+                    "*",
+                ],
+                "field1": {
+                    "ui:options": {"col": 4}
+                },
+                "field2": {
+                    "ui:options": {"col": 8}
+                },
+            }
+        )
```
```suggestion
        return json.dumps(
            {
                "ui:order": [
                    "field1",
                    "field2",
                    "*",
                ],
                "field1": {
                    "ui:options": {"col": 4}
                },
                "field2": {
                    "ui:options": {"col": 8}
                },
            }
        )
```

fabcor-maxiv commented on docs/source/dev/queue.md:
````diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+        )
+```
+
+**Note**: The `"ui:options.col"` key is mxcube-web specific, i.e. it is not defined in the `rjsf` library.
````
```suggestion
**Note**: The `"ui:options.col"` key is specific to MXCuBE-Web, i.e. it is not defined in the `rjsf` library.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
````diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+```
+
+**Note**: The `"ui:options.col"` key is mxcube-web specific, i.e. it is not defined in the `rjsf` library.
+These values correspond to `Bootstrap` `col-*` classes; i.e. these should be integers between 1 and 12 or "auto".
````
```suggestion
These values correspond to Bootstrap's `col-*` classes; i.e. these should be integers between 1 and 12 or `auto`.
```

fabcor-maxiv commented on docs/source/dev/queue.md:
```diff
@@ -205,10 +216,48 @@ class TestCollectionTaskParameters(BaseModel):
+
+**Note**: The `"ui:options.col"` key is mxcube-web specific, i.e. it is not defined in the `rjsf` library.
+These values correspond to `Bootstrap` `col-*` classes; i.e. these should be integers between 1 and 12 or "auto".
+Here's more about the [grid system](https://getbootstrap.com/docs/4.0/layout/grid/#grid-options)
```
```suggestion
Here's more about Bootstrap's [grid system](https://getbootstrap.com/docs/4.0/layout/grid/#grid-options)
```

mockoocy committed:
Documents DATA_MODEL ui-related methods

The documentation section, about `QueueEntry.DATA_MODEL` was missing
information about `update_dependent_fields` and `ui_schema` methods.
Also it gives insight about `ui:schema.col` option that has been
introduced in https://github.com/mxcube/mxcubeweb/pull/1657

Co-authored-by: fabcor <fabien.coronis@maxiv.lu.se>

mockoocy commented:
Thanks @fabcor-maxiv  for fixing the typos :D

I see that I've accidentally ran formatter on the file. It only added some whitespaces to the source file, so it should not be a big deal though.

fabcor-maxiv commented:
> I see that I've accidentally ran formatter on the file. It only added some whitespaces to the source file, so it should not be a big deal though.

Which formatter was used? It would be nice to add to our toolchain, if possible, so that we can improve consistency.

mockoocy commented:
> Which formatter was used? It would be nice to add to our toolchain, if possible, so that we can improve consistency.

Prettier, so the same as we use for formatting our js files in MXCuBE-web

fabcor-maxiv commented:
> > Which formatter was used? It would be nice to add to our toolchain, if possible, so that we can improve consistency.
> 
> Prettier, so the same as we use for formatting our js files in MXCuBE-web

OK, thanks. Maybe we should add it to our checkers, linters, and formatters somehow: <https://github.com/mxcube/mxcubecore/issues/1244>.

mockoocy commented:
I can't get the pipeline to work properly.
Perhaps it is casued, by `paths-ignore` clause in the `tests` workflow;
```yaml
    paths-ignore:
      - docs/**
```
This PR does not include changes outside of `docs/` directory. Maybe this causes this workflow not to run at all, but it is still expected by Github to run and finish succesfully?

Note: I managed to make the pipeline pass with some changes to a random python file; https://github.com/mxcube/mxcubecore/actions/runs/15462542393

marcus-oscarsson commented:
Its a bit strange, if your happy with the PR ill try to merge in anyways

mockoocy commented:
It'd be nice if you merge it.

fabcor-maxiv commented:
I knew I had complained about this already: <https://github.com/mxcube/mxcubecore/pull/979#issuecomment-2270767350> :)
